explore/src/main/scala/explore/tabs/ObsTabContents.scala

```diff
@@ -192,315 +190,293 @@ object ObsTabContents {
-    )(implicit ctx:     AppContextIO): VdomNode =
-      potRender[View[ObservationData]] {
-        Reuse.always { vod =>
-          val cgOpt: Option[ConstraintGroup] =
-            constraintGroups.get.find(_._1.contains(vod.get.id)).map(_._2)
-
-          Select(
-            value = cgOpt.map(cg => ObsIdSet.fromString.reverseGet(cg.obsIds)).orEmpty,
-            onChange = (p: Dropdown.DropdownProps) => {
-              val newCgOpt =
-                ObsIdSet.fromString
-                  .getOption(p.value.toString)
-                  .flatMap(ids => constraintGroups.get.get(ids))
-              newCgOpt.map { cg =>
-                vod.zoom(ObservationData.constraintSet).set(cg.constraintSet) >>
-                  ObsQueries
-                    .updateObservationConstraintSet[IO](List(vod.get.id), cg.constraintSet)
-                    .runAsyncAndForget
-              }.getOrEmpty
-            },
-            options = constraintGroups.get
-              .map(kv =>
-                new SelectItem(value = ObsIdSet.fromString.reverseGet(kv._1),
-                               text = kv._2.constraintSet.displayName
-                )
+  def makeConstraintsSelector(
+    constraintGroups: View[ConstraintsList],
+    obsView:          Pot[View[ObservationData]]
+  )(implicit ctx:     AppContextIO): VdomNode =
+    potRender[View[ObservationData]] {
+      Reuse.always { vod =>
+        val cgOpt: Option[ConstraintGroup] =
+          constraintGroups.get.find(_._1.contains(vod.get.id)).map(_._2)
+
+        Select(
+          value = cgOpt.map(cg => ObsIdSet.fromString.reverseGet(cg.obsIds)).orEmpty,
+          onChange = (p: Dropdown.DropdownProps) => {
+            val newCgOpt =
+              ObsIdSet.fromString
+                .getOption(p.value.toString)
+                .flatMap(ids => constraintGroups.get.get(ids))
+            newCgOpt.map { cg =>
+              vod.zoom(ObservationData.constraintSet).set(cg.constraintSet) >>
+                ObsQueries
+                  .updateObservationConstraintSet[IO](List(vod.get.id), cg.constraintSet)
+                  .runAsyncAndForget
+            }.getOrEmpty
+          },
+          options = constraintGroups.get
+            .map(kv =>
+              new SelectItem(value = ObsIdSet.fromString.reverseGet(kv._1),
+                             text = kv._2.constraintSet.displayName
               )
-              .toList
-          )
-        }
-      }(obsView)
-
-    // TODO Use this method
-    def readTargetPreferences(targetId: Target.Id)(implicit ctx: AppContextIO): Callback =
-      $.props.flatMap { p =>
-        p.userId.get.map { uid =>
-          UserTargetPreferencesQuery
-            .queryWithDefault[IO](uid, targetId, Constants.InitialFov)
-            .flatMap(v => $.modStateIn[IO](State.fovAngle.replace(v)))
-            .runAsyncAndForget
-        }.getOrEmpty
+            )
+            .toList
+        )
       }
+    }(obsView)
+
+  // TODO Use this method
```

Comment:
Why aren't we using this now?

Reply:
Because aladin has problems setting the fov on startup